src/main/kotlin/Calculator.kt

```diff
@@ -0,0 +1,58 @@
+import java.util.LinkedList
+import java.util.Queue
+
+object Calculator {
+
+    fun calEquation(equation: String?): Double {
+        if (equation == null || equation.isEmpty()) {
+            throw IllegalArgumentException("입력 값이 유효하지 않습니다.")
+        }
+
+        val (numList: Queue<Double>, operatorList) = LinkedList<Double>() to mutableListOf<Char>()
+        separateNumAndOpList(equation, numList, operatorList)
+
+        return calOperator(numList, operatorList)
+    }
+
+    fun calOperator(numList: LinkedList<Double>, operatorList: List<Char>): Double {
```

Comment:
인터페이스를 타입으로 사용하는 습관을 길러 두면 프로그램이 훨씬 유연해질 거예요. 인터페이스와 다형성에 대해 알아보고 아래의 글도 참고하면 좋을 것 같아요.
https://stackoverflow.com/questions/11683044/return-arraylist-or-list/11683073

```suggestion
    fun calOperator(numList: Queue<Double>, operatorList: List<Char>): Double {
```

Reply:
해당 블럭을 함수로 추출하는 기능을 쓰다보니 다형성 관련 부분을 체크하지 못했습니다.
감사합니다.:+1:
[8ea2b2c01a53a26c6e2d8d78201202d66b82e29a]